data/blog/2024/2024-08-05-using-laminas-continuous-integration.md

```diff
@@ -0,0 +1,113 @@
+You can check out the documentation on the page [GitHub Workflow Documentation](https://docs.github.com/en/actions/using-workflows/about-workflows).
+
+```yaml
+name: "Continuous Integration"
+
+on:
+  pull_request:
+  push:
+    branches:
+    tags:
+
+jobs:
+  ci:
+    uses: laminas/workflow-continuous-integration/.github/workflows/continuous-integration.yml@1.x
+```
+
+When pushing PHP code for review, Laminas CI generates a series of automated processes which are designed to:
+
+- Check **code style and formatting** using PHP Code Sniffer based on `phpcs.xml` (`phpcs.xml.dist` can also be used).
+- Run the battery of **tests** using PHPUnit based on `phpunit.xml` (`phpunit.xml.dist` can also be used).
+- Run **static analysis** to detect bugs and check that best programming practices are respected using psalm based on `psalm.xml` (`psalm.xml.dist` can also be used).
+- Run **yamllint and markdownlint** for `docs/book/**/*.md` files and `mkdocs.yml`.
+
+A full list of its capabilities is listed on [Laminas CI Automation](https://github.com/laminas/laminas-ci-matrix-action).
+Check it out to see what items it can identify based on the configuration files you include in your project.
+
+Based on your project and the files you update in a given push, you can expect to see a **job matrix** like the one below:
+
+- Documentation Linting [8.2, latest]
+- MkDocs Linting [8.2, latest]
+- PHPUnit [8.2, lowest]
+- PHPUnit [8.2, latest]
+- PHPUnit [8.3, lowest]
+- PHPUnit [8.3, latest]
+- PHPUnit [8.4, latest]
+- PHPCodeSniffer [8.2, latest]
+- Psalm [8.2, latest]
+
+Note that the job matrix generator is smart enough to ignore jobs that are not relevant to the affected items in your changes; if you only updated markdown files, then only markdown linting jobs will be run, or, if you've updated PHP code, your PHPUnit test suite will be run for all supported PHP versions listed in your `composer.json`.
+This ensures that the job matrix is as effective and efficient as possible, while still offering a solid review.
+
+Each of the above jobs takes its **configuration settings** straight from your project:
+
+- **PHP version**
+- **Additional extensions**
+- **php.ini settings**
+- **dependency** - lowest, locked, or latest
+
+Once the job has the above values, it **runs a command** which is the actual QA check.
+
+The automated process returns a result for each job.
+If you get any errors, the job is marked as failed.
+You may be able to fix some of the errors automatically in your local code e.g. via phpcs and psalm specific commands (phpcbf and psalter, respectively) while others need to be investigated manually.
+Only when all errors are resolved do you get the coveted green check mark. And a good night's sleep, of course.
+
+### Conclusions
+
+Laminas CI in a CI pipeline can be advantageous, especially for projects built with the Laminas//Mezzio microframework, but also for any other framework or library, given its broad range.
```

Comment:
> Laminas//Mezzio microframework

What is meant here and why two backslashes?

Reply:
It happens to use here and there 3 backslashes, as a reminder of laminas logo
Laminas///Mezzio
I rephrase it now

Reply:
> It happens to use here and there 3 backslashes, as a reminder of laminas logo

We have never done that.

Reply:
I did that in dotkernel related materials :-)